src/Aspire.Hosting/Dashboard/ResourceCollection.cs

```diff
@@ -47,30 +37,62 @@ void RemoveChannel(Channel<ResourceChange> channel)
         }
     }
 
-    private sealed class ChangeEnumerable : IAsyncEnumerable<ResourceChange>
+    /// <summary>
+    /// Integrates a changed resource within the cache, and broadcasts the update to any subscribers.
+    /// </summary>
+    /// <param name="resource">The resource that was modified.</param>
+    /// <param name="changeType">The change type (Added, Modified, Deleted).</param>
+    /// <returns>A task that completes when the cache has been updated and all subscribers notified.</returns>
+    public async ValueTask Integrate(ResourceViewModel resource, ObjectChangeType changeType)
```

Comment:
Could `ResourceViewModel` be renamed to `ResourceModel`? It doesn't have anything to do with views at this point.

Reply:
That's coming later. I've been doing the renames in stages. Ultimately it'll become an immutable object and I'm thinking of calling it `ResourceSnapshot`. View models will exist only in the dashboard.